### PR TITLE
Add ScopeReferenceProperty and UserReferenceProperty

### PR DESCRIPTION
**Pull request by raduiordache**
Fixes #<issue>

## Checklist:
- [x] My code follows the pykechain style
    - [x] I added type hinting to all functions
    - [x] I added documentation for all public functions
    - [x] I locally used `tox` to check for dists and docs errors
    - [x] My code passes the `pep8` and `flake8` linting checks and no warnings
    - [x] I removed unused imports, using `Code > Optimize Imports` in PyCharm
    - [x] I used [`black`](https://github.com/psf/black) to format the new code
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] I committed fresh test cassettes
    - [x] I have proper test coverage (don't decline the coverage of the test)
- [x] I asked another teammate to review the code
- [x] I update the Changelog.md with the appropriate changes.

### CODE REVIEW

**Comment by raduiordache** [comment]
@JelleBoersma @jberends - can you review this MR for me?

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,7 @@ v3.8.1-rc2 (UNRELEASED)
 
 * :+1: Added `set_associations` and `remove_associations` method to the `Widget` class (#827)
 * :+1: Renamed activity clone API endpoint is now supported. We now support the cloning and renaming of Parts as well as cloning the activities. (#805)
+* :+1: Added `UserReferenceProperty` and `ScopeReferenceProperty` classes (#832)
```
Comment (jberends) [inline CHANGELOG.rst:9]:
```suggestion
* :+1: Added support of user references and scope references properties in pykechain by implementing the `UserReferencesProperty` and `ScopeReferencesProperty` classes. (#832)
```

File: pykechain/models/__init__.py [hunk 2]
```diff
@@ -38,6 +38,8 @@
     'SelectListProperty2',
     'DatetimeProperty2'
     'ActivityReferenceProperty',
+    'ScopeReferenceProperty',
```
Comment (jberends) [inline pykechain/models/__init__.py:41]: This is a deviation from the naming in KE-chain. We call it UserReference`s`Property and ScopeReference`s`Property. That also holds for the ActivityReference`s`property though.

I suggest to re-align this naming in pykechain. As I look at it this is just inconsistent and sloppy.

May the `MultiReferenceProperty` is out of harms way as that one is already embedded being very old.

We need to discuss with @JelleBoersma on the impact on the code of scripts.

File: pykechain/models/__init__.py [hunk 3]
```diff
@@ -47,6 +49,8 @@
     PropertyType.REFERENCES_VALUE: MultiReferenceProperty2,
     PropertyType.DATETIME_VALUE: DatetimeProperty2,
     PropertyType.ACTIVITY_REFERENCES_VALUE: ActivityReferenceProperty,
+    PropertyType.SCOPE_REFERENCES_VALUE: ScopeReferenceProperty,
```
Comment (jberends) [inline pykechain/models/__init__.py:52]: you see, see last comment on ScopeReference**s**property

File: pykechain/models/property_reference.py [hunk 4]
```diff
@@ -25,3 +25,72 @@ def _retrieve_objects(self, **kwargs) -> List[Activity]:
             activity.refresh()  # To populate the object with all expected data
             activities.append(activity)
         return activities
+
+
+class ScopeReferenceProperty(_ReferencePropertyInScope):
+    """A virtual object representing a KE-chain Scope Reference property.
```
Comment (jberends) [inline pykechain/models/property_reference.py:31]:
```suggestion
    """A virtual object representing a KE-chain Scope References property.
```

File: pykechain/models/property_reference.py [hunk 5]
```diff
@@ -25,3 +25,72 @@ def _retrieve_objects(self, **kwargs) -> List[Activity]:
...
+                )
+
+        scopes = []
+        if scope_ids:
+            scopes = list(self._client.scopes(id__in=",".join(scope_ids)))
+        return scopes
+
+
+class UserReferenceProperty(_ReferencePropertyInScope):
+    """A virtual object representing a KE-chain User Reference property.
```
Comment (jberends) [inline pykechain/models/property_reference.py:65]:
```suggestion
    """A virtual object representing a KE-chain User References property.
```

**Review by jberends — CHANGES_REQUESTED** [review]
We need to change the naming of the property types in pykechain to align with the naming in KE-chain

File: pykechain/models/__init__.py (hunk 2 above)
Comment (raduiordache) [reply pykechain/models/__init__.py]: Discussed with @JelleBoersma , there is no impact on the ActivityReference property

**Review by jberends — APPROVED** [review]

**Comment by jberends** [comment]
This looks better indeed. Did you check with @JelleBoersma on the `ActivityReference_s_Property` refactoring. Impact is mainly when there is a assertion if the property is that specific classname. And I honestly think that if there are any assertions, they are performed on the `property_type` anyway.

**Comment by raduiordache** [comment]
> This looks better indeed. Did you check with @JelleBoersma on the `ActivityReference_s_Property` refactoring. Impact is mainly when there is a assertion if the property is that specific classname. And I honestly think that if there are any assertions, they are performed on the `property_type` anyway.

I have checked with @JelleBoersma , it appears there are no problems accordign to him.

**Pull request merged**